src/main/java/com/idrsolutions/microservice/JPedalServlet.java

```diff
@@ -150,17 +149,19 @@ protected void convert(Individual individual, Map<String, String[]> params,
         }
     }
 
-    private static void validateSettings(final Map<String, String> paramMap, final Mode mode) throws Exception {
-        final SettingsValidator settingsValidator = new SettingsValidator(paramMap);
+    @Override
+    protected SettingsValidator validateSettings(final String settings) {
+        final String[] conversionParams = settings != null ? getConversionParams(settings) : null;
+
+        final SettingsValidator settingsValidator = new SettingsValidator(conversionParams);
 
-        if (mode == Mode.convertToImages) {
-            settingsValidator.requireString("format", validEncoderFormats);
-            settingsValidator.optionalFloat("scaling", new float[]{0.1f, 10});
+        final String mode = settingsValidator.validateString("mode", validModes, true);
+        if (mode.equals(Mode.convertToImages.name())) {
```

Comment:
.validateString can return a null value, however this is not checked in the following if statement.

Recommend adding in a check like so:
```suggestion
        final String mode = settingsValidator.validateString("mode", validModes, true);
        if (mode != null && mode.equals(Mode.convertToImages.name())) {
```

Reply:
Changes made in the following commit